src/Microsoft.AspNetCore.Mvc.Api.Analyzers/ActualApiResponseMetadataFactory.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Mvc.Api.Analyzers
+{
+    public static class ActualApiResponseMetadataFactory
+    {
+        private static readonly Func<SyntaxNode, bool> _shouldDescendIntoChildren = ShouldDescendIntoChildren;
+
+        internal static bool TryGetActualResponseMetadata(
+            in ApiControllerSymbolCache symbolCache,
+            SemanticModel semanticModel,
+            MethodDeclarationSyntax methodSyntax,
+            CancellationToken cancellationToken,
+            out IList<ActualApiResponseMetadata> actualResponseMetadata)
+        {
+            actualResponseMetadata = new List<ActualApiResponseMetadata>();
+
+            var allReturnStatementsReadable = true;
```

Comment:
What's the significance of this? What happens if you run the codefix with a malformed return? Nothing?

Reply:
The code path we get this to be `false` is if Rolsyn says the returned `ITypeSymbol` is null or error. You get this if you have build errors (e.g. missing usings or references). We no-op in this case, just because we cannot accurately inspect action.

Reply:
I misremembered what this is for. We have the diagnostic (that's off by default), that tells you that you've documented a status code but haven't returned it. We use this as a flag to tell it not to do anything if we can't figure out the return type. Anyway, I added a test to verify we can print diagnostics for missing documentation when the action isn't complete (it's half written but compiles) and one for when it doesn't compile.